db/users.go

```diff
@@ -0,0 +1,85 @@
+package db
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/lbryio/lbrytv/config"
+	"github.com/lbryio/lbrytv/lbrynet"
+	"github.com/lbryio/lbrytv/models"
+	"github.com/lbryio/lbrytv/monitor"
+
+	"github.com/lbryio/lbry.go/extras/lbryinc"
+	"github.com/volatiletech/sqlboiler/boil"
+)
+
+// TokenHeader is the name of HTTP header which is supplied by client and should contain internal-api auth_token
+const TokenHeader string = "X-Lbry-Auth-Token"
+
+// GetUserByToken retrieves user by internal-api auth_token
+// TODO: Refactor out into logical pieces
+// TODO: Implement different error types for better error messages
+func GetUserByToken(token string) (*models.User, error) {
+	var u *models.User
+	ctx := context.Background()
+
+	u, err := models.Users(models.UserWhere.AuthToken.EQ(token)).OneG(ctx)
+	if err != nil {
+		Conn.Logger.LogF(monitor.F{"token": token}).Info("token supplied but isn't present in the database, trying internal-apis")
+		c := lbryinc.NewClient(token)
+		c.ServerAddress = config.Settings.GetString("InternalAPIHost")
```

Comment:
I would recommend that you have a const for these sort of hardcoded strings. This is a recipe for bugs a year from now when it gets changed and a search for the string is not performed, especially if there is reuse.

Reply:
see here https://github.com/lbryio/chainquery/blob/master/config/config.go#L28